compaction/compaction.cc

```diff
@@ -864,25 +878,93 @@ void compacted_fragments_writer::maybe_abort_compaction() {
     }
 }
 
-void compacted_fragments_writer::consume_new_partition(const dht::decorated_key& dk) {
+void compacted_fragments_writer::stop_current_writer() {
+    // stop sstable writer being currently used.
+    _stop_compaction_writer(&*_compaction_writer);
+    _compaction_writer = std::nullopt;
+}
+
+bool compacted_fragments_writer::can_split_large_partition() const {
+    return _c._can_split_large_partition;
+}
+
```

Comment:
I probably asked this already, but shouldn't this be done at the writer level?

Reply:
We had this discussion here: https://github.com/scylladb/scylladb/pull/11233#issuecomment-1208133226

TL;DR: we agreed it can live in compaction layer. I proposed a sstable_run_writer for the long-term.

Reply:
Ok